android/src/main/java/com/tailscale/ipn/MainActivity.kt

```diff
@@ -285,6 +293,34 @@ class MainActivity : ComponentActivity() {
     lifecycleScope.launch { Notifier.loginFinished.collect { _ -> loginQRCode.set(null) } }
   }
 
+  private fun showPermissionDeniedDialog() {
+    AlertDialog.Builder(this)
+        .setTitle(R.string.vpn_permission_denied)
+        .setMessage(R.string.multiple_vpn_explainer)
+        .setPositiveButton(R.string.go_to_settings) { _, _ ->
+          // Intent to open the VPN settings
+          val intent = Intent(Settings.ACTION_VPN_SETTINGS)
+          startActivity(intent)
+        }
+        .setNegativeButton(R.string.cancel, null)
+        .show()
+  }
+
+  fun isAnotherVpnActive(context: Context): Boolean {
+    val connectivityManager =
+        context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+
+    val activeNetwork = connectivityManager.activeNetwork
+    if (activeNetwork != null) {
+      val networkCapabilities = connectivityManager.getNetworkCapabilities(activeNetwork)
+      if (networkCapabilities != null &&
+          networkCapabilities.hasTransport(NetworkCapabilities.TRANSPORT_VPN)) {
+        return true
```

Comment:
Future improvement: I don't know if there is a way to know this, but it would be extremely useful to log the name of the other VPN app here. It would allow the support team to point the user to the app that is conflicting. Perhaps the active network contains a field with more information about the app that created it.

Reply:
agreed this would be useful, but unfortunately I don't think there is a good way to do this